lms/djangoapps/course_api/blocks/transformers/milestones.py

```diff
@@ -35,22 +44,79 @@ def collect(cls, block_structure):
         block_structure.request_xblock_fields('is_proctored_enabled')
         block_structure.request_xblock_fields('is_practice_exam')
         block_structure.request_xblock_fields('is_timed_exam')
+        block_structure.request_xblock_fields('entrance_exam_id')
+
+    def transform(self, usage_info, block_structure):
+        """
+        Modify block structure according to the behavior of milestones and special exams.
+        """
+
+        def add_special_exam_info(block_key):
+            """
+            Adds special exam information to course blocks.
+            """
+            if self.is_special_exam(block_key, block_structure):
+
+                #
+                # call into edx_proctoring subsystem
```

Comment:
I left this in place just because it's referring to `edx_proctoring` the package/repo.